### DISCUSSION TITLE
Delete Ending For Loop

### DISCUSSION THREAD

**Discussion by Seahorse1298** [nextgenhealthcare/connect#5080] [Q&A]
I am trying to write some code that iterates through a CCD message and is removing specific sections if the name of the entry contains the word "Hospital". An issue I am running in to is that it seems that the code will get down to the delete once and remove the entry, but after it gets rid of 1 entry it instantly kills the for loop and will not check for other entries with the "Hospital" name. Does anyone know how to solve this?

    var names = []

    for(i=0; i<numberOfSections; i++){
		    for(x=0; x<numberOfEntries); x++){						
			    if(msg['component']['structuredBody']['component'][i]['section']['entry'][x]['name'].toString().includes('Hospital')){
				    delete msg['component']['structuredBody']['component'][i]['section']['entry'][x]
			    }
		    }
	    }
    }

**Comment by jonbartels** [comment]
Some good examples are deleting OBX or NTE segments. https://forums.mirthproject.io/search?q=delete+obx+loop&searchJSON=%7B%22keywords%22%3A%22delete+obx+loop%22%7D 

Even though those are HL7v2 objects they still use an XML representation in MC from the e4x library. They use the same looping logic and the same delete logic. 

The filter logic shown here may also be useful https://github.com/nextgenhealthcare/connect-examples/tree/master/Code%20Templates/Filter%20XMLLists

**Reply by Seahorse1298** [reply]
The links are very useful! Seems like the delete is resetting the count to 0 thus I my for loop is looking for index ['1'] when it should be looking for 0. Gotta find a way now to build a work around that!

**Reply by pacmano1** [reply]
Another way to remove those sections is with XSLT.
